python/src/transaction.rs

```diff
@@ -123,7 +123,12 @@ impl FromPyObject<'_> for PyLance<Operation> {
                 let name = ob.getattr("name")?.extract()?;
                 let fields = ob.getattr("fields")?.extract()?;
                 let dataset_version = ob.getattr("dataset_version")?.extract()?;
-
+                let index_version: Option<String> = ob.getattr("index_version")?.extract()?;
+                let index_version =
+                    semver::Version::parse(&index_version.unwrap_or(String::from("0.27.0")))
```

Comment:
I don't think we can have a default here unless it isn't a constant.  This will be too difficult to update.  Should we just throw an error if the version is not set?